File: views/src/main/java/dlgdev/weighttracker/views/entry/NewWeightEntryView.java
package dlgdev.weighttracker.views.entry;

import android.content.Context;
import android.text.format.DateUtils;
import android.util.AttributeSet;
import android.widget.NumberPicker;
import android.widget.RelativeLayout;
import android.widget.TextView;

import org.joda.time.DateTime;

import butterknife.Bind;
import butterknife.ButterKnife;
import dlgdev.weighttracker.R;

public class NewWeightEntryView extends RelativeLayout {
	@Bind(R.id.date_view) TextView dateView;
	@Bind(R.id.value_view) NumberPicker valueView;

	public NewWeightEntryView(Context context) {
		super(context);
	}

	public NewWeightEntryView(Context context, AttributeSet attrs) {
		super(context, attrs);
	}

	public NewWeightEntryView(Context context, AttributeSet attrs, int defStyleAttr) {
		super(context, attrs, defStyleAttr);
	}

	public void setup(DateTime date, OnClickListener listener, int min, int max, double value) {
		ButterKnife.bind(this);
		setDate(date);
		dateView.setOnClickListener(listener);
		valueView.setMaxValue(max);
		valueView.setMinValue(min);
		valueView.setValue((int)value);
	}

	public void setDate(DateTime date) {
		dateView.setText(DateUtils.formatDateTime(getContext(), date.getMillis(),
				DateUtils.FORMAT_ABBREV_ALL | DateUtils.FORMAT_SHOW_YEAR));
	}

	public String getValue() {
		return Integer.toString(valueView.getValue());
	}
}

File: views/src/main/java/dlgdev/weighttracker/views/entry/NewWeightEntryView.kt
package dlgdev.weighttracker.views.entry

import android.content.Context
import android.text.format.DateUtils
import android.util.AttributeSet
import android.view.View
import android.widget.NumberPicker
import android.widget.RelativeLayout
import android.widget.TextView
import butterknife.BindView
import butterknife.ButterKnife
import dlgdev.weighttracker.R
import org.joda.time.DateTime

class NewWeightEntryView : RelativeLayout {
    @BindView(R.id.date_view) lateinit var dateView: TextView
    @BindView(R.id.value_view) lateinit var valueView: NumberPicker

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    fun setup(date: DateTime, min: Int, max: Int, value: Double, listener: View.OnClickListener) {
        ButterKnife.bind(this)
        setDate(date)
        dateView.setOnClickListener(listener)
        valueView.maxValue = max
        valueView.minValue = min
        valueView.value = value.toInt()
    }

    fun setDate(date: DateTime) {
        dateView.text = DateUtils.formatDateTime(context, date.millis,
                DateUtils.FORMAT_ABBREV_ALL or DateUtils.FORMAT_SHOW_YEAR)
    }

    val value: String
        get() = Integer.toString(valueView.value)
}
